lib/baza-rb.rb

```diff
@@ -579,6 +580,38 @@ def csrf
 
   private
 
+  # Update host from X-Zerocracy-Host header if present.
+  #
+  # @param [Typhoeus::Response] ret The HTTP response containing headers
+  # @param [Iri] uri The current URI object to update
+  # @return [Iri] The updated URI object (or original if no valid header present)
+  # @note Invalid hostnames are logged as warnings and ignored
+  def update_host_from_response(ret, uri)
```

Comment:
@paveltrty see https://www.yegor256.com/2018/08/22/builders-and-manipulators.html